### PR TITLE
feat(utils): add utils for WeChat MP

### PR DESCRIPTION
**Pull request by Rongronggg9**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/wechat/tgchannel/lifeweek
/wechat/feeds/Mzg2ODAyNTgyMQ==
/wechat/mp/homepage/MzA3MDM3NjE5NQ==/16
/wechat/mp/msgalbum/MzA3MDM3NjE5NQ==/1375870284640911361
/gzh360/category
/gzh360/gzh/北京青年报
/wechat/feeddd/6131e1441269c358aa0e2141
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
```
feat(*): add utils for WeChat MP

Motivation:
There are multiple routes that need to fetch articles from WeChat MP.
However, letting them fetch articles by themselves could potentially
lead to cache key collisions. Even if cache key collisions do not occur,
un-normalized URL could potentially lead to duplicated requests.
What's more, articles from WeChat MP have weird formats and need to be
fixed. Creating a universal function to do this work can create some
ease for new route contributors.

Note:
In order to make this PR atomic as much as possible, I did not touch
those broken routes. Once this PR is merged, I will try to fix them.

Signed-off-by: Rongrong <15956627+Rongronggg9@users.noreply.github.com>
```

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
It would be better if the annotations of the new utility can follow jsdoc

File: lib/utils/wechat-mp.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+/**
+ * Fetch article and its metadata from WeChat MP (mp.weixin.qq.com), then fill the `item` object with the result.
+ *
+ * If you use this function, no need to call `fetchArticle` or `fixArticleContent`
+ *
+ * A new route SHOULD use this function instead of manually calling the above functions
+ *
+ * An existing route adopting this function SHOULD either:
+ * - set `skipLink` to true (not recommended)
+ * - set `item.guid` to `item.link` BEFORE calling this function
```
Comment (TonyRL) [inline lib/utils/wechat-mp.js:134]: Could you explain why it's necessary to set `item.guid` to `item.link`? Middleware will handle it if no guid is set and guid isn't used in this utility.

https://github.com/DIYgod/RSSHub/blob/e0b7ca676d8b5817b474861ad674205444f3a15e/lib/views/rss.art#L35
https://github.com/DIYgod/RSSHub/blob/e0b7ca676d8b5817b474861ad674205444f3a15e/lib/views/atom.art#L37

Comment (Rongronggg9) [reply lib/utils/wechat-mp.js:134]: Only existing routes need to do that in order not to break existing subscriptions.

Different guid means different posts. RSS readers use guid to uniquely distinguish posts. A new guid will make the RSS reader consider the post as a never-seen post. Thus, it is necessary to make sure not to change the guid of the post of existing routes.

Comment (Rongronggg9) [reply lib/utils/wechat-mp.js:134]: This utility will normalize the URL. If not manually setting guid to the original URL, the guid will be changed and make RSS readers consider an ever-seen post as never-seen.

Comment (TonyRL) [reply lib/utils/wechat-mp.js:134]: Understood.

**Pull request merged**